refinery/core/api.py

```diff
@@ -831,6 +843,54 @@ def get_study_assays(self, request, **kwargs):
             study_uuid=kwargs['study_uuid']
         )
 
+    def handle_incomplete_dataset(self, dataset):
+        """
+        Handle an incomplete DataSet instance when a detail view is
+        accessed.
+        :param dataset: a DataSet object
+        :return: a DataSet object
+        :raises: A TastyPie NotFound Exception if said DataSet
+        doesn't have the proper information associated yet.
+        """
+        if dataset.get_version_details() is not None:
+            return dataset
+
+        else:
+            logger.error(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+            raise NotFound(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+
+    def handle_incomplete_datasets(self, datasets):
+        """
+        Handle incomplete DataSet instances when a list view is accessed.
+        :param datasets: a list of DataSet objects
+        :return: a list of DataSet objects sans ones that don't have the
+        proper information associated yet.
+        """
+        complete_datasets = []
+
+        for dataset in datasets:
+            if dataset.get_version_details() is None:
```

Comment:
Probably even better to add a validation method to the `DataSet` class that would return a boolean. Then you can simply use that in `get_obj()` and `get_object_list()` instead of `handle_incomplete_dataset()`. It could also be easily re-used elsewhere.